src/workflow.ts

```diff
@@ -80,6 +80,143 @@ export async function startWorkflow() {
+            initial: false,
+          },
+        ],
+        {
+          onCancel: () => {
+            throw new Error(red("✖") + " Operation cancelled");
+          },
+        }
+      );
+
+      if (confirm) {
+        // choose the option prompt
+        const changeOptions = await prompts(
+          [
+            {
+              type: "select",
+              name: "optionToChange",
+              message: "Select the option you want to change",
+              choices: [
+                { title: "Project Name", value: "projectName" },
+                { title: "Template", value: "template" },
+                { title: "Network", value: "network" },
+                { title: "Package Manager", value: "packageManager" },
+              ],
+            },
+          ],
+          {
+            onCancel: () => {
+              throw new Error(red("✖") + " Operation cancelled");
+            },
+          }
+        );
+
+        // Ask for new values based on the user's choice and update the result object
+        switch (changeOptions.optionToChange) {
+          case "projectName":
+            result.projectName = (
+              await prompts({
+                type: "text",
+                name: "projectName",
+                message: "Enter new project name",
+                initial: result.projectName,
+                validate: (value) => validateProjectName(value),
+              })
+            ).projectName;
+            break;
+          case "template":
+            result.template = (
+              await prompts({
+                type: "select",
+                name: "template",
+                message: "Choose how to start",
+                choices: [
+                  {
+                    title: "Dapp Boilerplate",
+                    value: "dapp-boilerplate",
+                    description:
+                      "A simple and light-weight web based dapp template that includes the basic structure needed for starting a dapp",
+                  },
+                  {
+                    title: "Node Boilerplate",
+                    value: "node-boilerplate",
+                    description:
+                      "A simple and light-weight node template that includes the basic structure needed for starting a node project on Aptos",
+                  },
+                  {
+                    title: "Todolist dapp",
+                    value: "todolist-boilerplate",
+                    description:
+                      "A fully working todo list dapp with pre-implemented smart contract and UI",
+                  },
+                ],
+                initial: 0,
+              })
+            ).template;
+            break;
+          case "network":
+            result.network = (
+              await prompts({
+                type: "select",
+                name: "network",
+                message: "Choose your network",
```

Comment:
we have a repeated code for each prompt object - let's get each step object a global variable and use it in the initial flow and on the rechoce flow

Reply:
@0xmaayan great review. I wanted to make all these changes initially but wasn't sure if you would like this format. Before I resolve this last conversation, just making sure you check out the workflowOptions object inside the rechoseWorkflow.ts file. If you have any other recommendations let me know!